### ISSUE TITLE
[FEATURE] Support HTTPStatus

### ISSUE THREAD

**Issue by hmvp** [tiangolo/fastapi#1349]
### Is your feature request related to a problem

We typically use [HTTPStatus](https://docs.python.org/3/library/http.html#http.HTTPStatus) in our code. When used as a value for `status_code` in path functions this results in the string literal in the documentation and when "trying" the response is shown as undocumented because it doesn't match the actual response code

### The solution you would like

I want to use HTTPStatus enum values and have it work exactly like the `starlette.status` pseudo enum values

**Comment by retnikt** [comment]
For now you can create your own HTTPStatus class (you can't subclass Enums) and add a `__str__` method to convert it to the string representation of the actual status code, for example:
```python
class MyHTTPStatus:
    OK = 200
    ...
    def __str__(self):
        return str(int(self))
```
or use `int(HTTPStatus)` in the endpoint decorator.

**Comment by tiangolo** [comment]
Thanks for the help @retnikt ! :bow: 

Yeah, @hmvp you can probably use that for your use case.

**Comment by hmvp** [comment]
I am sorry to disagree...

It is indeed quite easy to workaround this. For example by just importing status from starlette and using that in the decorators. 

However `HTTPStatus` is part of the standard lib and since we use it extensively in the rest of our code I would like to use that instead of the starlette status. The second reason is that I expected it to work (since its part of stdlib) and it somewhat did but gave unexpected results.. This is a papercut/pitfall/surpising behavior of fastapi and given the high standard of the rest of the library this should not be there 

I am not sure if @retnikt  understood what I am trying to do here.. Adding another class is a weird suggestion given that both `HTTPStatus` and `starlette.status` already exist

**Comment by tiangolo** [comment]
The ASGI spec uses `int`s for status codes, not enums. That's why it takes `int`s.

Also, @retnikt has been helping a lot here, answering a lot of questions, trying to help others like you, for free. Just out of being a cool person. Please try to be more respectful to the community that is here trying to help.

If you really want to use `HTTPStatus` enums you can use the `int` value, it's quite simple, e.g. `HTTPStatus.OK.value`.

**Comment by hmvp** [comment]
I did not want to be disrespectful to @retnikt and I value his contribution for other people that might have a similar but different issue. I just don't see how it is relevant for my issue. I am sorry if my wording to strong... As a non-native English speaker I might miss some subtleties.

With regards to the ASGI spec, I did not know that, but I also was not aware that the `status_code` code argument followed the ASGI spec in that regard, especially since putting in a HTTPStatus code enum value just works, but will give a weird result in the docs. It should not be to difficult to add some code along the line of:
```
if isinstance(HTTPStatus, status_code):
     status_code = status_code.value
```
or  just `status_code = int(status_code)` to the path functions, which would solve a papercut and would still be valid ASGI.  Otherwise I would expect the path functions to be noisy about wrong input..

On the other hand, this is indeed not a big issue and if you don't want to change anything, thats fine with me. I just wanted to signal that this is a [papercut](https://en.wikipedia.org/wiki/Paper_cut_bug) in fastapi

**Comment by tiangolo** [comment]
Cool, thanks!

Yeah, I would accept a PR checking if a status code is an enum to get its value first. :nerd_face: :heavy_check_mark:

**Comment by retnikt** [comment]
Working on it!

**Comment by tiangolo** [comment]
This was added on https://github.com/tiangolo/fastapi/pull/1534 by @retnikt :rocket: :tada: 

It will be available in FastAPI version `0.57.0` (released in a couple of hours).

**Comment by hmvp** [comment]
Works as expected :+1:
